Shellspec can not collect data with the profile option enabled

Hello,
I try to use the profile option, but the formatter get no proper data.

```
# shellspec --profile spec/em_spec.sh
Running: /bin/bash [bash 5.1.4(1)-release]
............/lib/shellspec/lib/general.sh: line 538: * 10: syntax error: operand expected (error token is "* 10")
Aborted with status code [executor: 0] [reporter: 0] [error handler: 102]
Fatal error occurred, terminated with exit status 102.
```
This is a problem in shellspec_shift10() because "$2" is empty.
Therefore, I try find the way the function will be called.

But this ended because I can not find the code where "$time_real" filled with proper values.
See /lib/shellspec/lib/libexec/reporter/profiler_formatter.sh in profiler_end() in the line
```
  read_profiler callback "$profiler_tick_total" "$time_real" \
    < "$SHELLSPEC_PROFILER_LOG" \
    > "$SHELLSPEC_PROFILER_REPORT"
```
The time_real is empty, is ''. If I assign "0" to the empty "$time_real" I get no useful report but Shellspec finished without an error.

My question, where the variable "$time_real" came from?

Sorry, I can not find, what is going wrong,
Lutz

p.s.
I use shellspec 0.28.1 dist version.

Hello Luke,
you are right, as long as the US number format is used everything works well.
But sometimes (in some environments, I switched to LANG=en_US), the output from the used "time" command looks different to the expected output.

The output (in a AIX environment) evaluated by read_time_log() echoed to the terminal (see lines prefixed by "#### ").
```
# shellspec --profile spec/em_spec.sh   Running: /bin/bash [bash 5.1.16(1)-release]
............#### /home/lutz/lib/shellspec/libexec/shellspec-executor.sh: line 9: [: : integer expression expected
####
#### Real   6.86
#### User   0.82
#### System 0.46
#### /home/lutz/lib/shellspec/libexec/shellspec-executor.sh: line 9: [: : integer expression expected
####
#### Real   6.86
#### User   0.82
#### System 0.46


Finished in 6.86 seconds (user 0.82 seconds, sys 0.46 seconds)
12 examples, 0 failures

# Top 10 slowest examples of the 12 examples
#   1 0.7171 spec/em_spec.sh:66-83
#   2 0.6268 spec/em_spec.sh:36-42
#   3 0.6197 spec/em_spec.sh:101-105
#   4 0.5499 spec/em_spec.sh:52-58
#   5 0.5480 spec/em_spec.sh:44-50
#   6 0.5409 spec/em_spec.sh:60-64
#   7 0.5192 spec/em_spec.sh:87-92
#   8 0.4966 spec/em_spec.sh:8-12
#   9 0.4908 spec/em_spec.sh:14-18
#  10 0.4787 spec/em_spec.sh:29-34
```
And sometimes (in a Linux environment), the output evaluated looks like this (see lines prefixed by "#### ").
```
> shellspec --profile spec/em_spec.sh
Running: /bin/sh [bash 4.3.48(1)-release]
............#### /usr/local/lib/shellspec/libexec/shellspec-executor.sh: line 9: [: : integer expression expected
#### real 42.09
#### user 1.63
#### sys 0.67
#### /usr/local/lib/shellspec/libexec/shellspec-executor.sh: line 9: [: : integer expression expected
#### real 42.09
#### user 1.63
#### sys 0.67


Finished in 42.09 seconds (user 1.63 seconds, sys 0.67 seconds)
12 examples, 0 failures

# Top 10 slowest examples of the 12 examples
#   1 30.2448 spec/em_spec.sh:66-83
#   2 10.1672 spec/em_spec.sh:87-92
#   3 0.1492 spec/em_spec.sh:36-42
#   4 0.1449 spec/em_spec.sh:44-50
#   5 0.1439 spec/em_spec.sh:52-58
#   6 0.1416 spec/em_spec.sh:101-105
#   7 0.1375 spec/em_spec.sh:14-18
#   8 0.1361 spec/em_spec.sh:8-12
#   9 0.1346 spec/em_spec.sh:60-64
#  10 0.1306 spec/em_spec.sh:22-27
```
Therefore, my suggested patch to handle the output use some additional lines of code, to handle Linux and AIX environments and your change.
```
--- shellspec/lib/libexec/reporter.sh.orig      2021-01-11 06:58:16.000000000 +0100
+++ shellspec/lib/libexec/reporter.sh   2024-02-02 10:58:14.827350190 +0100
@@ -17,15 +17,25 @@
# $1: prefix, $2: filename
read_time_log() {
   eval "$1_real='' $1_user='' $1_sys=''"
   [ -r "$2" ] || return 1
   # shellcheck disable=SC2034
   while IFS= read -r line; do
-    case $line in (real[\ $TAB]*|user[\ $TAB]*|sys[\ $TAB]*)
-      case ${line##*[ $TAB]} in (*[!0-9.]*) continue; esac
-      eval "$1_${line%%[ $TAB]*}=\"\${line##*[ \$TAB]}\""
+    case $line in ([Rr]eal[\ $TAB]*|[Uu]ser[\ $TAB]*|[Ss]ys[\ $TAB]*|[Ss]ystem[\ $TAB]*)
+      case ${line##*[ $TAB]} in (*[!0-9.,]*) continue; esac
+      case $line in
+        [Rr]eal[\ $TAB]*)
+          eval "$1_real=\"\${line##*[ \$TAB]}\"";;
+        [Uu]ser[\ $TAB]*)
+          eval "$1_user=\"\${line##*[ \$TAB]}\"";;
+        [Ss]ys[\ $TAB]*)
+          eval "$1_sys=\"\${line##*[ \$TAB]}\"";;
+        [Ss]ystem[\ $TAB]*)
+          eval "$1_sys=\"\${line##*[ \$TAB]}\"";;
+        *) eval "$1_${line%%[ $TAB]*}=\"\${line##*[ \$TAB]}\"";;
+      esac
     esac
   done < "$2" &&:
   eval "[ \"\$$1_real\" ] && [ \"\$$1_user\" ] && [ \"\$$1_sys\" ]"
}

field_description() {
```
Based on your suggested patch (handle '.' and ',') and some additional lines Shellspec is able to profile the scripts in Linux, AIX (and MacOS) enviroments, in some more/bad weather conditions.

Nice, thanks for your investigations,
Lutz

--- a/lib/libexec/reporter.sh
+++ b/lib/libexec/reporter.sh
@@ -22,7 +22,7 @@ read_time_log() {
   # shellcheck disable=SC2034
   while IFS= read -r line; do
     case $line in (real[\ $TAB]*|user[\ $TAB]*|sys[\ $TAB]*)
-      case ${line##*[ $TAB]} in (*[!0-9.]*) continue; esac
+      case ${line##*[ $TAB]} in (*[!0-9.,]*) continue; esac
       eval "$1_${line%%[ $TAB]*}=\"\${line##*[ \$TAB]}\""
     esac
   done < "$2" &&:

